crates/red_knot_test/src/parser.rs

```diff
@@ -164,8 +164,12 @@ static HEADER_RE: LazyLock<Regex> =
 static CODE_RE: LazyLock<Regex> = LazyLock::new(|| {
     Regex::new(
         r"(?x)
-        ^```(?<lang>(?-u:\w)+)?(?<config>(?:\x20+\S+)*)\s*\n
-        (?<code>(?:.|\n)*?)\n?
+        ^(?:
+            `(?<path>[^`\n]+)`\s*:\s*\n
```

Comment:
The mandatory colon looks like something that's easy to miss. Should we make it optional in the pattern here, capture whether it's present or not, and then show an error if it's missing? Otherwise, if someone forgets to use one, it might be more difficult to debug what's wrong.

Reply:
I think we should do something about that. That's a disadvantage of specifying the path outside the code snippet. Someone might write something like:

````
A long sentence that forces a line break .... this also works with
`int`:
```py
…
```
````

And it looks like we would currently treat this as being a code block for a file named `int`. I guess we could enforce two newlines before `<path>:`?

Reply:
Yes, I think requiring two consecutive newlines before the path would reduce the potential for error, without ruling out any formatting we would want to use; I don't think we'd ever want the file path to be closely visually associated with the preceding line/paragraph.